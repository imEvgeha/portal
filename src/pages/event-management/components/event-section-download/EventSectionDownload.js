import React from 'react';
import PropTypes from 'prop-types';
import Button from '@atlaskit/button';
import {download} from '../../../../util/Common';
import {DOWNLOAD, JSON_MIME} from '../../eventManagementConstants';

const EventSectionDownload = ({data, filename}) => {
    const handleDownload = () => {
        download(JSON.stringify(data), filename, JSON_MIME);
    };

    return (
        <Button onClick={handleDownload}>{DOWNLOAD}</Button>
    );
};

EventSectionDownload.propTypes = {
    data: PropTypes.object.isRequired,
    filename: PropTypes.string.isRequired,
};

export default EventSectionDownload;
