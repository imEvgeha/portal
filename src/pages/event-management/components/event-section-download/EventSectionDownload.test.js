import React from 'react';
import {shallow} from 'enzyme';
import EventSectionDownload from './EventSectionDownload';

describe('EventSectionDownload', () => {
    let wrapper;
    beforeEach(() => {
        const props = {
            data: { test: 'awesome string'},
            filename: 'event_filename',
        };
        wrapper = shallow(<EventSectionDownload {...props} />);
    });

    it('should render the component', () => {
        expect(wrapper).not.toBe(null);
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
