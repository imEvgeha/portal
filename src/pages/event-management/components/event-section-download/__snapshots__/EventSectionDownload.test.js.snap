// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventSectionDownload should match snapshot 1`] = `
<WithAnalyticsContext(WithAnalyticsEvents(Button))
  onClick={[Function]}
>
  Download
</WithAnalyticsContext(WithAnalyticsEvents(Button))>
`;
